lib/improver/tests/spotdata_new/spotdata_new/test_NeighbourSelection.py

```diff
@@ -0,0 +1,641 @@
+                                             x_points, y_points)
+        radius = self.global_orography.coord_system().semi_major_axis
+        component = radius/np.sqrt(2.)
+        sub_component = component/np.sqrt(2.)
+        expected = [[sub_component, -sub_component, -component]]
+
+        self.assertArrayAlmostEqual(result, expected)
+
+
+class Test_build_KDTree(Test_NeighbourSelection):
+
+    """Test construction of a KDTree with scipy."""
+
+    def test_basic(self):
+        """Test that the expected number of nodes are created and that a tree
+        is returned; this should be the lengths of the x and y coordinates
+        multiplied in the simple case."""
+
+        plugin = NeighbourSelection()
+        result, result_nodes = plugin.build_KDTree(self.region_land_mask)
+        expected_length = (self.region_land_mask.shape[0] *
+                           self.region_land_mask.shape[1])
+
+        self.assertEqual(result_nodes.shape[0], expected_length)
+        self.assertIsInstance(result, scipy.spatial.ckdtree.cKDTree)
+
+    def test_only_land(self):
+        """Test that the expected number of nodes are created and that a tree
+        is returned. In this case the number of nodes should be this should be
+        equal to the number of land points."""
+
+        plugin = NeighbourSelection(land_constraint=True)
+        result, result_nodes = plugin.build_KDTree(self.region_land_mask)
+        expected_length = np.nonzero(self.region_land_mask.data)[0].shape[0]
+
+        self.assertEqual(result_nodes.shape[0], expected_length)
+        self.assertIsInstance(result, scipy.spatial.ckdtree.cKDTree)
+
+
+class Test_select_minimum_dz(Test_NeighbourSelection):
+
+    """Test extraction of the minimum height difference points from a provided
+    array of neighbours."""
+
+    def test_basic(self):
+        """Test a simple case where the first element in the provided lists
+        has the smallest vertical displacement to the site. Expect the
+        coordinates of the first node to be returned."""
+
+        plugin = NeighbourSelection()
+        site_altitude = 3.
+        nodes = np.array([[0, 0], [1, 0], [2, 0], [3, 0], [4, 0]])
+        distance = np.arange(5)
+        indices = np.arange(5)
+
+        result = plugin.select_minimum_dz(self.region_orography,
+                                          site_altitude, nodes,
+                                          distance, indices)
+        self.assertArrayEqual(result, nodes[0])
+
+    def test_some_invalid_points(self):
+        """Test a case where some nodes are beyond the imposed search_radius,
+        which means they have a distance of np.inf, ensuring this is handled.
+        Also change the site height so the second node is the expected
+        result."""
+
+        plugin = NeighbourSelection()
+        site_altitude = 5.
+        nodes = np.array([[0, 0], [1, 0], [2, 0], [3, 0], [4, 0]])
```

Comment:
This test isn't working. I suspect the nodes are wrong.

Reply:
Yup, I changed the orography due to a great circle issue and forgot to update this. Fixed.